src/Microsoft.Xaml.Behaviors/Behavior.cs

```diff
@@ -47,7 +47,7 @@ public abstract class Behavior :
         private Type associatedType;
         private DependencyObject associatedObject;
 
-        internal event EventHandler AssociatedObjectChanged;
+        public event EventHandler AssociatedObjectChanged;
```

Comment:
Why would you need this to be public? If you need to know when the associated object changed, you can just override the `OnAttached` and `OnDettaching` methods.

Reply:
An example of its use can be found in https://github.com/MartinEgli/XamlBehaviorsWpf/tree/Behaviors.Extensions/src/Behaviors.Extensions . The extension is in a separate project. A class similar to EventTriggerBase should be created that inherits from TriggerBase and the class should behave similarly. Access to internal elements is required to create a class DependencyPropertyChangedEventTriggerBase. 

The class DependencyPropertyChangedEventTriggerBase inherits from TriggerBase. With the method OnAttached newBehavior.AssociatedObjectChanged += this.OnBehaviorHostChanged becomes identical to the class EventTriggerBase. To enable access to AssociatedObjectChanged the visibility must be changed to public. The same for behavior.AssociatedObjectChanged -= this.OnBehaviorHostChanged.